crates/via-router/src/visitor.rs

```diff
@@ -103,41 +107,52 @@ impl<'a, 'b, T> Visitor<'a, 'b, T> {
+        let is_exact_match = self.path.get(depth + 1).is_none();
         let path_segment_value = &self.path.value[start..end];
 
         for index in node.entries() {
             let next = &self.routes[*index];
 
-            if path_segment_value != next.pattern {
-                // The path segment does not match the pattern of the next node.
-                // We can skip this node and continue to search for a match.
-                continue;
-            }
-
-            if let Pattern::CatchAll(_) = next.pattern {
-                // The end offset of `path_segment` should be the end offset
-                // of the last path segment in the url path.
-                let path_segment_range = self.path.slice_from(depth);
-
-                // The next node has a `CatchAll` pattern and will be considered
-                // an exact match. Due to the nature of `CatchAll` patterns, we
-                // do not have to continue searching for descendants of this
-                // node that match the remaining path segments.
-                self.matches.push(Match {
-                    path_segment_range,
-                    is_exact_match: true,
-                    node: next,
-                });
-            } else {
-                self.matches.push(Match {
-                    path_segment_range,
-                    is_exact_match: self.path.get(depth + 1).is_none(),
-                    node: next,
-                });
-
-                // Continue to match descendants of `next` against the path
-                // segment at the next depth.
-                self.match_at_depth(depth + 1, next);
+            match next.pattern {
+                Pattern::CatchAll(_) => {
+                    // The next node has a `CatchAll` pattern and will be considered
+                    // an exact match. Due to the nature of `CatchAll` patterns, we
+                    // do not have to continue searching for descendants of this
+                    // node that match the remaining path segments.
+                    self.matches.push(Match {
+                        // The end offset of `path_segment` should be the end offset
+                        // of the last path segment in the url path.
+                        path_segment_range: self.path.slice_from(depth),
+                        is_exact_match: true,
+                        node: next,
+                    });
+                }
+                Pattern::Dynamic(_) => {
+                    self.matches.push(Match {
+                        path_segment_range,
+                        is_exact_match,
+                        node: next,
+                    });
+
+                    // Continue to match descendants of `next` against the path
+                    // segment at the next depth.
+                    self.match_at_depth(depth + 1, next);
+                }
+                Pattern::Static(value) if value == path_segment_value => {
```

Comment:
It would be nice to remove the duplication between the match arms of `Pattern::Static` and `Pattern::Dynamic. However, I'm concerned that would result in unnecessary overhead or make the code more difficult to follow.